pkgs/development/python-modules/support.nix

```diff
@@ -0,0 +1,80 @@
+{ pkgs, python }:
+
+with pkgs.lib;
+
+self: rec {
+
+  inherit python;
+
+  pythonAtLeast = versionAtLeast python.pythonVersion;
+  pythonOlder = versionOlder python.pythonVersion;
+  isPy26 = python.majorVersion == "2.6";
+  isPy27 = python.majorVersion == "2.7";
+  isPy33 = python.majorVersion == "3.3";
+  isPy34 = python.majorVersion == "3.4";
+  isPy35 = python.majorVersion == "3.5";
+  isPy36 = python.majorVersion == "3.6";
+  isPyPy = python.executable == "pypy";
+  isPy3k = strings.substring 0 1 python.majorVersion == "3";
+
+  callPackage = pkgs.newScope self;
+
+  bootstrapped-pip = callPackage ./bootstrapped-pip { };
+
+  buildPythonPackage = makeOverridable (callPackage ./generic {
+    inherit bootstrapped-pip;
+  });
+
+  buildPythonApplication = args: buildPythonPackage ({namePrefix="";} // args );
+
+  # Unique python version identifier
+  pythonName =
+    if isPy26 then "python26" else
+    if isPy27 then "python27" else
+    if isPy33 then "python33" else
+    if isPy34 then "python34" else
+    if isPy35 then "python35" else
+    if isPy36 then "python36" else
+    if isPyPy then "pypy" else "";
```

Comment:
`throw "<some descriptive error message>"` would be better in this case. But could we perhaps just use `python.name` (which should be something like "python-2.6" etc)? The problem with hardcoding `pythonName` here is that it is difficult for people to add a new interpreter locally (say they have packaged `python3.7` and want to test this out), because they have to edit nixpkgs to add a new interpreter.